File: app/src/main/java/com/billiamram/hexagonalscaffold/ExampleComponent.java
package com.billiamram.hexagonalscaffold;

import com.billiamram.main.CoreModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {
        AndroidModule.class,
        AndroidAdaptersModule.class,
        CoreModule.class
})
public interface ExampleComponent {
    void inject(ExampleActivity exampleActivity);
}

File: app/src/main/java/com/billiamram/hexagonalscaffold/ExampleComponent.kt
package com.billiamram.hexagonalscaffold

import com.billiamram.main.CoreModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidModule::class, AndroidAdaptersModule::class, CoreModule::class))
interface ExampleComponent {
    fun inject(exampleActivity: ExampleActivity)
}
